src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -478,61 +479,74 @@ def build_container_from_source_with_buildpack(self, image_name, source):  # pyl
-        if buildpack_run_tag is not None:
-            buildpack_run_image = f"mcr.microsoft.com/oryx/builder:{buildpack_run_tag}"
-            logger.debug(f"Determined the run image to use as {buildpack_run_image}.")
-            command.extend(['--run-image', buildpack_run_image])
-
         # If the user specifies a target port, pass it to the buildpack
         if self.target_port:
             command.extend(['--env', f"PORT={self.target_port}"])
 
-        logger.debug(f"Calling '{' '.join(command)}'")
-        try:
-            is_non_supported_platform = False
-            with subprocess.Popen(command, stdout=subprocess.PIPE) as process:
+        could_build_image = False
+        for builder_image in builder_image_list:
+            # Run 'pack build' to produce a runnable application image for the Container App
+            # Specify the image as the 'build-cache' image to ensure that the local cache is used for build layers
+            command = [pack_exec_path, 'build', cache_image_name, '--builder', builder_image, '--path', source, '--tag', image_name]
 
-                # Stream output of 'pack build' to warning stream
-                while process.stdout.readable():
-                    line = process.stdout.readline()
-                    if not line:
-                        break
+            logger.debug(f"Calling '{' '.join(command)}'")
+            try:
+                is_non_supported_platform = False
+                is_non_supported_os = False
+                with subprocess.Popen(command, stdout=subprocess.PIPE) as process:
 
-                    stdout_line = str(line.strip(), 'utf-8')
-                    logger.warning(stdout_line)
-                    if not is_non_supported_platform and "No buildpack groups passed detection" in stdout_line:
-                        is_non_supported_platform = True
+                    # Stream output of 'pack build' to warning stream
+                    while process.stdout.readable():
+                        line = process.stdout.readline()
+                        if not line:
+                            break
 
-                # Update the result of process.returncode
-                process.communicate()
-                if is_non_supported_platform:
-                    raise ValidationError("Current buildpacks do not support the platform targeted in the provided source code.")
+                        stdout_line = str(line.strip(), 'utf-8')
+                        logger.warning(stdout_line)
 
-                if process.returncode != 0:
-                    raise CLIError("Non-zero exit code returned from 'pack build'; please check the above output for more details.")
+                        # Check if the application is targeting a platform that's found in the current builder
+                        if not is_non_supported_platform and "No buildpack groups passed detection" in stdout_line:
+                            is_non_supported_platform = True
```

Comment:
Could you please confirm: should we leave early the while loop when this happens? Any reasons to stick with it or can we directly move to the next builder? Same comment below

Reply:
I'm going to keep this as-is for now -- typically, these messages are going to be at the end of the process anyway, so there wouldn't be much more to process before the flow moves on to the next builder, but more importantly, I'm not entirely sure what the ramifications of exiting the process early are without first calling `process.communicate()`. I can test this out to be sure that it won't affect the next builder call, but I also don't think it would make a difference performance-wise if we kept it as-is.